src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -272,13 +323,22 @@ private static void addFieldDefinition(@NotNull GoLiteralValue literalValue, @No
     private final GoCompositeLit myCompositeLit;
     private final GoAssignmentStatement myAssignment;
     private final List<GoReferenceExpression> myReferenceExpressions;
+    private final GoStatement myStructDeclaration;
+    private final String myStructVarName;
+    private final boolean myNeedReplaceDeclarationWithShortVar;
 
     public Data(@NotNull GoAssignmentStatement assignment,
-                @NotNull GoCompositeLit compositeLit,
-                @NotNull List<GoReferenceExpression> referenceExpressions) {
+                @Nullable GoCompositeLit compositeLit,
+                @NotNull List<GoReferenceExpression> referenceExpressions,
+                @Nullable GoStatement structDeclaration,
+                @Nullable String structVarName,
+                boolean needReplaceDeclarationWithShortVar) {
```

Comment:
I think you can get rid of `needReplaceDeclarationWithShortVar` field by giving null compositeLit